src/components/withCurrentUserPersonalDetails.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {ComponentType, RefAttributes, ForwardedRef, useMemo} from 'react';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import getComponentDisplayName from '../libs/getComponentDisplayName';
+import ONYXKEYS from '../ONYXKEYS';
+import personalDetailsPropType from '../pages/personalDetailsPropType';
+import type {PersonalDetails, Session} from '../types/onyx';
+
+type CurrentUserPersonalDetails = PersonalDetails | Record<string, never>;
+
+type WithCurrentUserPersonalDetailsProps = {
+    currentUserPersonalDetails: CurrentUserPersonalDetails;
+};
+
+type WithCurrentUserPersonalDetailsComponentProps = {
+    /** Personal details of all the users, including current user */
+    personalDetails: Record<string, PersonalDetails> | null;
+
+    /** Session of the current user */
+    session: OnyxEntry<Session>;
+};
+
+// TODO: remove when all components that use it will be migrated to TS
+const withCurrentUserPersonalDetailsPropTypes = {
+    currentUserPersonalDetails: personalDetailsPropType,
+};
+
+const withCurrentUserPersonalDetailsDefaultProps: WithCurrentUserPersonalDetailsProps = {
+    currentUserPersonalDetails: {},
+};
+
+export default function <TProps extends WithCurrentUserPersonalDetailsProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>) {
```

Comment:
Add return type for this function, similar how Bartek did [here](https://github.com/Expensify/App/pull/29426/files)

Reply:
@blazejkustra Since the return value is wrapped into withOnyx it will look like this
```suggestion
export default function <TProps extends WithCurrentUserPersonalDetailsProps, TRef>(
    WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>,
): ComponentType<Omit<Omit<TProps, keyof WithCurrentUserPersonalDetailsProps> & WithCurrentUserPersonalDetailsComponentProps, keyof WithCurrentUserPersonalDetailsComponentProps>> {
```
I'm not sure if it makes sense to add it there since withOnyx hoc is typed, so it automatically detects type.
What do you think?